src/platform/Linux/NetworkCommissioningWiFiDriver.cpp

```diff
@@ -123,96 +170,247 @@ void LinuxWiFiDriver::Shutdown()
 CHIP_ERROR LinuxWiFiDriver::CommitConfiguration()
 {
     auto & kvs = PersistedStorage::KeyValueStoreMgr();
-    ReturnErrorOnFailure(kvs.Put(kWiFiSSIDKeyName, mStagingNetwork.ssid, mStagingNetwork.ssidLen));
-#if CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
-    if (mStagingNetwork.UsingPDC())
+
+    for (uint8_t networkIndex = 0; networkIndex < kMaxNetworks; networkIndex++)
```

Comment:
Shouldn't this only go up to `mCurrentNetId` or whatever the actually initialized number is?  Otherwise it seems like we are persisting garbage data?

Reply:
Yes will make more sense to only go upto `mCurrentNetId`

Reply:
Deleting stale entries would make sense, yes.